sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -327,11 +327,27 @@ private[joins] object UnsafeHashedRelation {
     // Create a mapping of buildKeys -> rows
     val keyGenerator = UnsafeProjection.create(key)
     var numFields = 0
+    val nullPaddingCombinations: Seq[UnsafeProjection] = if (isNullAware) {
```

Comment:
Can you add a small example here to illustrate what this block of code is doing. What do you think of extracting this into a method for better readability ? It's a bit big to be inline. Perhaps take an example of like 3 keys.

Reply:
Also do we use nullPaddingCombinations when all the keys are null ?